src/contrib/mlir/lowerer.cpp

```diff
@@ -336,6 +338,56 @@ namespace
         rewriter.replaceOp(op, {result});
     }
 
+    // ADD
+    REWRITER(NGReluOp)
+    {
+        auto loc = cast<NGReluOp>(op).getLoc();
+
+        auto result = m_pass.buildOutputDefs(op, rewriter)[0];
+        NGRAPH_CHECK(result->getType().isa<MemRefType>());
+        // Note that builder's current function is still the original function body.
+        // use getBlock to get the new block instead.
+
+        // get new operands
+        Value* lhs = operands[0];
+
+        ScopedContext scope(rewriter, loc);
+        // Views
+        MemRefView vRes(result), vLHS(lhs);
+        // Index Values
+        IndexedValue iRes(result), iLHS(lhs);
+        // Bounds Index Handles
+        auto lbs = vLHS.getLbs();
+        auto ubs = vLHS.getUbs();
+        // Loop induction vars
+        auto ivs = IndexHandle::makeIndexHandles(vLHS.rank());
+        auto pivs = IndexHandle::makeIndexHandlePointers(ivs);
+        // Steps
+        auto steps = vLHS.getSteps();
+
+        NGRAPH_CHECK(lhs->getType().isa<MemRefType>());
+        Type elemTy = lhs->getType().dyn_cast<MemRefType>().getElementType();
+        NGRAPH_CHECK(!elemTy.isa<FloatType>(),
+                     "NGReluOp with float element type should not be lowered until MLIR supports "
+                     "lowering !std.CmpF");
+
+        // clang-format off
+        LoopNestBuilder(pivs, lbs, ubs, steps)(
+            // single stmt body
+            [&] {
+                    if (auto floatTy = elemTy.dyn_cast<FloatType>()) {
```

Comment:
I thought comparison intrinsics are there. Why not just do 
`iRes(ivs) = intrinsics::select(iLHS(ivs) < zero, iLHS(ivs), zero);`

Reply:
iLHS(ivs) is of type TemplatedIndexValue and we dont have overloaded comparisons for that. https://github.com/tensorflow/mlir/blob/master/include/mlir/EDSC/Helpers.h#L217-#L258

Reply:
Comparison operators exist for `ValueHandle` though, and `TemplateIndexValue` has a cast operator to `ValueHandle`. You should be able to use the operator. 
```
ValueHandle mlir::edsc::op::operator<(ValueHandle lhs, ValueHandle rhs) {
  // TODO(ntv,zinenko): signed by default, how about unsigned?
  return createComparisonExpr(CmpIPredicate::SLT, lhs, rhs);
}
```